pawnpy/__init__.py

```diff
@@ -167,82 +181,107 @@ class AMX():
     def __init__(self, filename):
         self._filename = filename
         self._amx = AMXNative()
-        self._public_functions = dict()  # name:str -> index:number
 
         result = lib.aux_LoadProgram(
             byref(self._amx), c_char_p(filename.encode('utf-8')), 0)
         if result != AMX_ERR_NONE:
-            raise Exception('aux_LoadProgram failed with code %d' % result)
-
-        numNatives = c_int()
-        result = lib.amx_NumNatives(byref(self._amx), byref(numNatives))
-        if result != AMX_ERR_NONE:
-            raise Exception('amx_NumNatives failed with code %d' % result)
+            raise RuntimeError('aux_LoadProgram failed with code %d' % result)
 
-        print(numNatives)
+        self._publics = None
+        self.__init_publics()
 
-        if numNatives.value != 0:
-            name = create_string_buffer(sNAMEMAX + 1)
-            result = lib.amx_GetNative(byref(self._amx), 0, name, None)
-            if result != AMX_ERR_NONE:
-                raise Exception('amx_GetNative failed with code %d' % result)
+        self._natives = None
+        self.__init_natives()
 
-            print(name.value)
+        self._callbackptr = AMX_CALLBACK(self._callback)
```

Comment:
Why do you need it?

Reply:
To save the wrapper - otherwise it will be collected

Reply:
No, I meant why you setting callback at all? I I remember correctly that's like advanced topic and we don't need it unless we want to change behavior

Reply:
You're right - I should use one or the other - let's stick with direct natives hookup